### ISSUE TITLE
Wrappers around aleju/imgaug?

### ISSUE THREAD

**Issue by ppwwyyxx** [ppwwyyxx/tensorpack#399]
https://github.com/aleju/imgaug seems to have very nice augmentors. And the library even have the same name as our augmentors!

Maybe we could provide a wrapper on the interface to make it easy to use those augmentors in tensorpack. One thing is that they only support uint8 images.

@vfdev-5 any comments? since you seem to have experience with both libraries.

**Comment by vfdev-5** [comment]
Yes, there are nice features as `iaa.Sometimes`, `iaa.SomeOf` etc (similar to `RandomApplyAug` and `RandomChooseAug`), keypoints transformations, `iaa.ElasticTransformation` and `iaa.PiecewiseAffine`, and other color augmentations, dropouts etc. Image augmentation randomness is implemented in a [developed way](https://github.com/aleju/imgaug/blob/master/imgaug/parameters.py), various sampling provided.  
However, it mostly uses `skimage` vs `opencv` (in tensorpack) and bounded by `uint8` and range `0-255`.  In my experience, transformations like `iaa.PiecewiseAffine` can slow down dataflow.

@ppwwyyxx I do not know what kind of wrapping you would like to provide. Probably, the simplest way to use `aleju/imgaug` within tensorpack is with `MapData`. And it would be up to user to handle type of transformations, sampling, determinism/repeatability of `aleju/imgaug` transformations. 

HTH

**Comment by ppwwyyxx** [comment]
What I had in mind is a MapData as well.

Just found that the keypoints transformations in iaa seems to assume integer coordinates -- this is going to hurt training for coordinate sensitive tasks.
